app/helpers/audits_helper.rb

```diff
@@ -47,13 +49,18 @@ def details(audit)
         end
       end
     elsif !main_object? audit
-      ["#{audit_action_name(audit).humanize} #{id_to_label audit.audited_changes.keys[0], audit.audited_changes.values[0]}
-       #{audit_action_name(audit)=="removed" ? "from" : "to"} #{audit.associated_name || id_to_label(audit.audited_changes.keys[1], audit.audited_changes.values[1])}"]
+      ["#{audit_action_name(audit).humanize} #{id_to_label(audit.audited_changes.keys[0], audit.audited_changes.values[0], owner_type(audit))}
+       #{audit_action_name(audit)=="removed" ? "from" : "to"} #{audit.associated_name || id_to_label(audit.audited_changes.keys[1], audit.audited_changes.values[1], owner_type(audit))}"]
     else
       []
     end
   end
 
+
+  def owner_type(audit)
```

Comment:
@isratrade, helpers should be used for DRY-ing UI templates. Very classic example for this rule is the `audit_time` function, but the `owner_type` is actually part of the model (the audit object) as it exposes more semantics out of it.
The same goes for `audit_template?` and `audit_login?`.